app/models/differential_expression_result.rb

```diff
@@ -13,22 +13,31 @@ class DifferentialExpressionResult
     DEFAULT_COMP_METHOD, 'logreg', 't-test', 't-test_overestim_var'
   ].freeze
 
+  # analysis types, e.g. A vs. B and A vs. All
+  DEFAULT_ANALYSIS = 'a-vs-all'.freeze
+  ANALYSIS_TYPES = [DEFAULT_ANALYSIS, 'a-vs-b'].freeze
```

Comment:
(If we don't end up requiring separate uploads for pairwise and one-vs-all, then this comment might be moot.)

A few weeks ago, I recall the team came to rough consensus on referring to "A vs. B" as "pairwise", to avoid confusion with A/B testing.  This would also align SCP vocabulary with [diffxpy](https://diffxpy.readthedocs.io/en/latest/api/index.html#multiple-tests-per-gene), and some other scientific works.

I'd also suggest we use the most community-aligned terms for other kinds of DE analysis.  Scanpy, diffxpy, and at least some [publications](https://www.biorxiv.org/content/10.1101/2022.05.09.490241v1.full.pdf) basically replace "all" with "rest" (which actually seems more accurate), e.g. "one-vs-rest" or "versus-rest".

```suggestion
  # analysis types, e.g. pairwise (a.k.a. A vs. B) and one-vs-rest (a.k.a. A vs. all)
  DEFAULT_ANALYSIS = 'one-vs-rest'.freeze
  ANALYSIS_TYPES = [DEFAULT_ANALYSIS, 'pairwise'].freeze
```

Reply:
Ah yes you are correct!  Thanks.

Reply:
> (If we don't end up requiring separate uploads for one-vs-rest and pairwise, then this comment might be moot.)

My feeling is that even if we allow a unified upload that has both pairwise and one-vs-rest in the same file, there may be a benefit to keeping track of this at the result level as there potentially will be a slightly different UX when showing the results.  If we decide later that this isn't needed, it's trivial to remove the data with a migration.

Reply:
[922ab25](https://github.com/broadinstitute/single_cell_portal_core/pull/1753/commits/922ab25a49483d5e2c0aee0c5667fedf8a79a4d7)